exchanges/order/orders.go

```diff
@@ -547,83 +653,86 @@ func (d *Detail) InferCostsAndTimes() {
 	}
 }
 
-// FilterOrdersBySide removes any order details that don't match the
-// order status provided
+// FilterOrdersBySide removes any order details that don't match the order
+// status provided
 func FilterOrdersBySide(orders *[]Detail, side Side) {
-	if side == "" || side == AnySide {
+	if bypassSideFilter&side == side || len(*orders) == 0 {
 		return
 	}
 
-	var filteredOrders []Detail
+	target := 0
 	for i := range *orders {
-		if strings.EqualFold(string((*orders)[i].Side), string(side)) {
-			filteredOrders = append(filteredOrders, (*orders)[i])
+		if (*orders)[i].Side == side {
+			(*orders)[target] = (*orders)[i]
+			target++
 		}
 	}
-
-	*orders = filteredOrders
+	*orders = (*orders)[:target]
 }
 
 // FilterOrdersByType removes any order details that don't match the order type
 // provided
 func FilterOrdersByType(orders *[]Detail, orderType Type) {
-	if orderType == "" || orderType == AnyType {
+	if bypassTypeFilter&orderType == orderType || len(*orders) == 0 {
 		return
 	}
 
-	var filteredOrders []Detail
+	target := 0
 	for i := range *orders {
-		if strings.EqualFold(string((*orders)[i].Type), string(orderType)) {
-			filteredOrders = append(filteredOrders, (*orders)[i])
+		if (*orders)[i].Type == orderType {
+			(*orders)[target] = (*orders)[i]
+			target++
 		}
 	}
-
-	*orders = filteredOrders
+	*orders = (*orders)[:target]
 }
 
 // FilterOrdersByTimeRange removes any OrderDetails outside of the time range
-func FilterOrdersByTimeRange(orders *[]Detail, startTime, endTime time.Time) {
-	if startTime.IsZero() ||
-		endTime.IsZero() ||
-		startTime.Unix() == 0 ||
-		endTime.Unix() == 0 ||
-		endTime.Before(startTime) {
-		return
+func FilterOrdersByTimeRange(orders *[]Detail, startTime, endTime time.Time) error {
+	if len(*orders) == 0 {
+		return nil
+	}
+
+	if err := common.StartEndTimeCheck(startTime, endTime); err != nil {
+		if errors.Is(err, common.ErrStartAfterEnd) {
+			return fmt.Errorf("cannot filter orders by time range %w", err)
+		}
+		return nil
```

Comment:
Why are you returning nil here? I think I take back a previous nit. Perhaps just have:
```
if startTime.After(endTime) {
		return fmt.Errorf("cannot filter orders by time range %w", common.ErrStartAfterEnd)
	}
```
Because it seems the other scenarios could still potentially filter orders rather than returning